### PR TITLE
r/aws_sagemaker_hub_content_reference: new resource

### PR DESCRIPTION
**Pull request by subham-ibmhc**
## Rollback Plan

If a change needs to be reverted, we will publish an updated version of the library.

## Changes to Security Controls

Are there any changes to security controls (access controls, encryption, logging) in this pull request? If so, explain.

### Description

Adds a new resource for sagemaker service, `hub_content_reference` along with RI and List support.

### Relations

Closes https://hashicorp.atlassian.net/browse/FRB-7785

### References

https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_CreateHubContentReference.html
https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_UpdateHubContentReference.html
https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_DescribeHubContent.html
https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_DeleteHubContentReference.html
https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ListHubs.html
https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ListHubContents.html

### Output from Acceptance Testing

```console
% make t T=TestAccSageMakerHubContentReference_ K=sagemaker
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Running acceptance tests on branch: 🌿 f-sagemaker-hub-content-reference 🌿...
TF_ACC=1 go1.26.3 test ./internal/service/sagemaker/... -v -count 1 -parallel 20 -run='TestAccSageMakerHubContentReference_'  -timeout 360m -vet=off
2026/06/11 23:56:53 Creating Terraform AWS Provider (SDKv2-style)...
2026/06/11 23:56:53 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccSageMakerHubContentReference_Identity_basic
=== PAUSE TestAccSageMakerHubContentReference_Identity_basic
=== RUN   TestAccSageMakerHubContentReference_Identity_regionOverride
=== PAUSE TestAccSageMakerHubContentReference_Identity_regionOverride
=== RUN   TestAccSageMakerHubContentReference_basic
=== PAUSE TestAccSageMakerHubContentReference_basic
=== RUN   TestAccSageMakerHubContentReference_disappears
=== PAUSE TestAccSageMakerHubContentReference_disappears
=== RUN   TestAccSageMakerHubContentReference_minVersion
=== PAUSE TestAccSageMakerHubContentReference_minVersion
=== RUN   TestAccSageMakerHubContentReference_tags
=== PAUSE TestAccSageMakerHubContentReference_tags
=== CONT  TestAccSageMakerHubContentReference_Identity_basic
=== CONT  TestAccSageMakerHubContentReference_disappears
=== CONT  TestAccSageMakerHubContentReference_basic
=== CONT  TestAccSageMakerHubContentReference_tags
=== CONT  TestAccSageMakerHubContentReference_minVersion
=== CONT  TestAccSageMakerHubContentReference_Identity_regionOverride
--- PASS: TestAccSageMakerHubContentReference_disappears (29.24s)
--- PASS: TestAccSageMakerHubContentReference_basic (41.53s)
--- PASS: TestAccSageMakerHubContentReference_Identity_regionOverride (45.51s)
--- PASS: TestAccSageMakerHubContentReference_Identity_basic (48.50s)
--- PASS: TestAccSageMakerHubContentReference_tags (74.80s)
--- PASS: TestAccSageMakerHubContentReference_minVersion (78.17s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/sagemaker  83.333s
...
```

```console
% make t T=TestAccSageMakerHubContentReference_ K=sagemaker
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Running acceptance tests on branch: 🌿 f-sagemaker-hub-content-reference 🌿...
TF_ACC=1 go1.26.3 test ./internal/service/sagemaker/... -v -count 1 -parallel 20 -run='TestAccSageMakerHubContentReference_'  -timeout 360m -vet=off
2026/06/11 20:54:31 Creating Terraform AWS Provider (SDKv2-style)...
2026/06/11 20:54:31 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccSageMakerHubContentReference_basic
=== PAUSE TestAccSageMakerHubContentReference_basic
=== RUN   TestAccSageMakerHubContentReference_disappears
=== PAUSE TestAccSageMakerHubContentReference_disappears
=== RUN   TestAccSageMakerHubContentReference_minVersion
=== PAUSE TestAccSageMakerHubContentReference_minVersion
=== RUN   TestAccSageMakerHubContentReference_tags
=== PAUSE TestAccSageMakerHubContentReference_tags
=== CONT  TestAccSageMakerHubContentReference_basic
=== CONT  TestAccSageMakerHubContentReference_minVersion
=== CONT  TestAccSageMakerHubContentReference_tags
=== CONT  TestAccSageMakerHubContentReference_disappears
--- PASS: TestAccSageMakerHubContentReference_disappears (28.59s)
--- PASS: TestAccSageMakerHubContentReference_basic (32.34s)
--- PASS: TestAccSageMakerHubContentReference_minVersion (75.19s)
--- PASS: TestAccSageMakerHubContentReference_tags (78.26s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/sagemaker  83.623s
```

### CODE REVIEW

**Comment by subham-ibmhc** [comment]
@copilot+claude-sonnet-4.6 Generate a changelog for this PR.

File: internal/service/sagemaker/hub_content_reference.go [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+		_, err := conn.UpdateHubContentReference(ctx, &input)
+		if err != nil {
+			smerr.AddError(ctx, &response.Diagnostics, err, smerr.ID, new.HubContentName.ValueString())
+			return
+		}
+	}
+
+	smerr.AddEnrich(ctx, &response.Diagnostics, response.State.Set(ctx, new))
+}
+
```
[the remark this answers is not in the document]
Comment (YakDriver) [reply internal/service/sagemaker/hub_content_reference.go:286]: copilot-feedback:actionable

I think Copilot is right, you need to wait + flatten (update timeout), like create

File: internal/service/sagemaker/hub_content_reference.go [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+						},
+						"Removing min_version requires replacement of the resource.",
+						"Removing min_version requires replacement of the resource.",
+					),
+				},
+			},
+			"sagemaker_public_hub_content_arn": schema.StringAttribute{
+				CustomType:  fwtypes.ARNType,
+				Required:    true,
+				Description: "ARN of the public JumpStart hub content to reference.",
```
Comment (YakDriver) [inline internal/service/sagemaker/hub_content_reference.go]: These descriptions should be the same as in the documentation.

File: internal/service/sagemaker/hub_content_reference.go [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+
+	r.SetDefaultCreateTimeout(15 * time.Minute)
+	r.SetDefaultDeleteTimeout(15 * time.Minute)
+
+	return r, nil
+}
+
+const (
+	ResNameHubContentReference     = "Hub Content Reference"
+	hubContentReferenceIDPartCount = 2
```
Comment (YakDriver) [inline internal/service/sagemaker/hub_content_reference.go]: seems to be unused

File: internal/service/sagemaker/status.go [hunk 4]
```diff
@@ -255,6 +255,22 @@ func statusHub(conn *sagemaker.Client, name string) retry.StateRefreshFunc {
 	}
 }
 
+func statusHubContent(conn *sagemaker.Client, hubName, hubContentName string) retry.StateRefreshFunc {
```
Comment (YakDriver) [inline internal/service/sagemaker/status.go]: nit: Should this be called statusHubContentReference?

File: internal/service/sagemaker/hub_content_reference.go [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+		return
+	}
+
+	if _, err := waitHubContentReferenceDeleted(ctx, conn, hubName, hubContentName, r.DeleteTimeout(ctx, data.Timeouts)); err != nil {
+		smerr.AddError(ctx, &response.Diagnostics, err, smerr.ID, hubContentName)
+	}
+}
+
+// stripARNVersion removes the version suffix from a SageMaker ARN
+func stripARNVersion(arn *string) *string {
```
Comment (YakDriver) [inline internal/service/sagemaker/hub_content_reference.go:323]: could use a unit test for edge cases

File: internal/service/sagemaker/hub_content_reference.go [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+	"github.com/hashicorp/terraform-provider-aws/internal/retry"
+	"github.com/hashicorp/terraform-provider-aws/internal/smerr"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	inttypes "github.com/hashicorp/terraform-provider-aws/internal/types"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// @FrameworkResource("aws_sagemaker_hub_content_reference", name="Hub Content Reference")
+// @Tags(identifierAttribute="hub_content_arn")
```
Comment (YakDriver) [inline internal/service/sagemaker/hub_content_reference.go:40]: might be good to explain in a comment that the stripped arn is used for tags - might be confusing to a someone looking at this

File: internal/service/sagemaker/hub_content_reference.go [hunk 7]
```diff
@@ -0,0 +1,384 @@
...
+				Computed:    true,
+				Description: "ARN of the private SageMaker Hub that contains the content reference.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"hub_content_arn": schema.StringAttribute{
+				CustomType:  fwtypes.ARNType,
+				Computed:    true,
+				Description: "ARN of the hub content reference.",
```
Comment (YakDriver) [inline internal/service/sagemaker/hub_content_reference.go]: should be the same as the website docs and any arns that use stripped version should probably have some explanation

**Review by YakDriver — COMMENTED** [review]
For the majority, I think this looks good. Just a few small things.

File: internal/service/sagemaker/hub_content_reference.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (subham-ibmhc) [reply internal/service/sagemaker/hub_content_reference.go:286]: Added in [2179596](https://github.com/hashicorp/terraform-provider-aws/pull/48379/commits/21795960c2902b92e249f0b5eb2a90ef90449e3b)

File: internal/service/sagemaker/hub_content_reference.go (hunk 5 above)
Comment (subham-ibmhc) [reply internal/service/sagemaker/hub_content_reference.go:323]: Great catch. This would make the ARN malformed if there was no version suffix. Refactored this in [b689e28](https://github.com/hashicorp/terraform-provider-aws/pull/48379/commits/b689e28281dba514722e32530a47db5a985dd371) to match the version suffix pattern and replace it with `""` instead.

File: internal/service/sagemaker/hub_content_reference.go (hunk 6 above)
Comment (subham-ibmhc) [reply internal/service/sagemaker/hub_content_reference.go:40]: Added both as a comment to the resource file and in the documentation and description for the attributes.

**Review by YakDriver — APPROVED** [review]
LGTM 🎉 

```console
=== CONT  TestAccSageMakerHubContentReference_Identity_basic
=== CONT  TestAccSageMakerHubContentReference_basic
=== CONT  TestAccSageMakerHubContentReference_minVersion
=== CONT  TestAccSageMakerHubContentReference_disappears
=== CONT  TestAccSageMakerHubContentReference_List_includeResource
=== CONT  TestAccSageMakerHubContentReference_List_regionOverride
=== CONT  TestAccSageMakerHubContentReference_List_basic
=== CONT  TestAccSageMakerHubContentReference_Identity_regionOverride
=== CONT  TestAccSageMakerHubContentReference_tags
--- PASS: TestAccSageMakerHubContentReference_List_regionOverride (22.84s)
--- PASS: TestAccSageMakerHubContentReference_disappears (25.37s)
--- PASS: TestAccSageMakerHubContentReference_List_includeResource (26.90s)
--- PASS: TestAccSageMakerHubContentReference_List_basic (32.91s)
--- PASS: TestAccSageMakerHubContentReference_Identity_regionOverride (33.03s)
--- PASS: TestAccSageMakerHubContentReference_basic (33.63s)
--- PASS: TestAccSageMakerHubContentReference_Identity_basic (39.10s)
--- PASS: TestAccSageMakerHubContentReference_tags (56.39s)
--- PASS: TestAccSageMakerHubContentReference_minVersion (62.21s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/sagemaker	69.390s
```
